modular_splurt/code/datums/interactions/lewd/_lewd.dm

```diff
@@ -238,6 +238,13 @@
 				H.mob_climax(TRUE, "masturbation", "none")
 			else
 				H.mob_climax(TRUE, "sex", partner, !cumin, target_gen)
+	if(iscarbon(src))
+		var/mob/living/carbon/carbon_src
+
+		carbon_src = src
+		for(var/obj/item/organ/genital/G in carbon_src.internal_organs)
+			if(check_orgasm(sender = G))
+				break
```

Comment:
Wouldn't this call `check_orgasm` twice since you're also calling it in `do_climax` with the signal?

Reply:
eek! sorry